### PR TITLE
Variables: Add support for `$__timezone` template variable

### PR DESCRIPTION
**Pull request by VictorColomb**
### What is this feature?

This PR adds a new `$__timezone` variables, replaced by the currently selected time zone by the template server (in queries, labels, etc.)

### Why do we need this feature?

 Use case: Using the Postgres datasource, it allows using functions such as `DATE_TRUNC` to manipulate datetimes in the correct timezone server-side. Example: `DATE_TRUNC('month', TIMEZONE('$__timezone', my_datetime_column))`. See #38159.

Bonus use case: Synchronizing time zones between Grafana and the [clock-panel plugin](https://github.com/grafana/clock-panel).

### Who is this feature for?

All Grafana OSS & Enterprise users.

### Which issue(s) does this PR fix?

Not an issue, but:

Fixes #38159

### Special notes for your reviewer:

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/state/time.test.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ describe('Explore item reducer', () => {
       });
       await dispatch(updateTime({ exploreId: ExploreId.left }));
       expect(mockTimeSrv.init).toBeCalled();
-      expect(mockTemplateSrv.updateTimeRange).toBeCalledWith(MOCK_TIME_RANGE);
+      expect(mockTemplateSrv.updateTimeRange).toBeCalledWith(MOCK_TIME_RANGE, undefined);
```
Comment (VictorColomb) [inline public/app/features/explore/state/time.test.ts:42]: FYI, with `defaultTimeZone` from `@grafana/schema`, my first rewrite of this test was as follows. It failed.

```js
expect(mockTemplateSrv.updateTimeRange).toBeCalledWith(MOCK_TIME_RANGE, defaultTimeZone);
```

**Review by torkelo — CHANGES_REQUESTED** [review]
This should be implemented as a macro (like __data, __value, __url_time_range) 

https://github.com/grafana/grafana/blob/main/public/app/features/templating/macroRegistry.ts

We will turn all the global vars like __user, __dashboard and __org into macros

**Comment by torkelo** [comment]
Looks good, now we just need to add a section here https://github.com/grafana/grafana/blob/main/docs/sources/dashboards/variables/add-template-variables/index.md#__rate_interval

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 2]
```diff
@@ -315,6 +315,12 @@ This is used in several places, including:
 - SQL queries in MySQL, Postgres, and MSSQL.
 - The `$__timeFilter` variable is used in the MySQL data source.
 
+### $\_\_timezone
+
+The `$__timezone` variable returns the currently selected time zone, either `utc` or an entry of the IANA time zone database (e.g. `America/New_York`).
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:320]: We avoid Latin abbreviations.
```suggestion
The `$__timezone` variable returns the currently selected time zone, either `utc` or an entry of the IANA time zone database (for example, `America/New_York`).
```

File: docs/sources/dashboards/variables/add-template-variables/index.md [hunk 3]
```diff
@@ -315,6 +315,12 @@ This is used in several places, including:
...
+
+If the currently selected time zone is _Browser Time_, your browser time zone will be guessed.
```
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:322]:
```suggestion
If the currently selected time zone is _Browser Time_, Grafana will try to determine your browser time zone.
```

File: docs/sources/dashboards/variables/add-template-variables/index.md (hunk 2 above)
Comment (imatwawana) [inline docs/sources/dashboards/variables/add-template-variables/index.md:320]: Is there any specific reason that _UTC_ isn't capitalized on line 320?

**Review by imatwawana — COMMENTED** [review]
Hi @VictorColomb, thanks for the contribution! I've made some wording suggestions and left a question.

**Review by imatwawana — APPROVED** [review]
Docs LGTM!

**Review by torkelo — APPROVED** [review]

**Pull request merged**